app/Services/Installment/InstallmentService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Services\Installment;
+
+use App\Models\Installment;
+use App\Repositories\Installment\InstallmentRepositoryInterface;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Pagination\LengthAwarePaginator;
+
+class InstallmentService {
+    public function __construct(private InstallmentRepositoryInterface $installmentRepository)
+    {}
+
+    public function create(array $installment): void {
+        $this->installmentRepository->create($installment);
+    }
+
+    public function findById(int $id): Installment {
+        return $this->installmentRepository->findById($id);
+    }
+
+    public function update(array $payload, Installment $installment): void {
+        $installment->update($payload);
+    }
+
+    public function determineStatusInstallment (array $installmentRequest, $id): array {
+        $installment = $this->findById($id);
+        if(doubleval($installmentRequest['paid_amount']) > 0  && doubleval($installmentRequest['paid_amount']) < $installment->amount) {
+            $installmentRequest['status'] = 'P';
+            $this->update($installmentRequest, $installment);
+            return $installmentRequest;
+        } else if (doubleval($installmentRequest['paid_amount']) === $installment->amount) {
+            $installmentRequest['status'] = 'B';
+            $this->update($installmentRequest, $installment);
+            return $installmentRequest;
+        } else {
+            $installmentRequest['status'] = 'A';
+            $this->update($installmentRequest, $installment);
+            return $installmentRequest;
+        }
```

Comment:
Creio que se você está utilizando _else-if_ quer dizer que você não conhece o conceito de early return. 

[Post para auxiliar](https://dev.to/danielhe4rt/tres-conceitos-para-um-codigo-limpo-3mdo)